### ISSUE TITLE
labels.csv: support path-based image resolution, not stem-only

### ISSUE THREAD

**Issue by stanlrt** [CAIIVS/raitap#95]
## Summary

`labels.csv` IDs are normalised to filename **stem** only (`Path(text).stem` at `src/raitap/data/data.py:483,505`). This collapses rows whose stems collide across class subdirectories and forces consumers with nested `ImageFolder`-style layouts (`test/NORMAL/*.jpg`, `test/PNEUMONIA/*.jpg`) to build a parallel flat directory just for RAITAP. Make `image` column accept relative paths resolved against `data.source`.

## Why this matters

Standard PyTorch / torchvision practice is `ImageFolder` layout: one subdir per class. Many public datasets (Kaggle Chest X-Ray Pneumonia, ImageNet, CIFAR-as-images) ship this way. Filenames are not guaranteed unique across class subdirs (e.g. `IM-0001.jpeg` can appear in both `NORMAL/` and `PNEUMONIA/`), so stem-only matching silently drops samples.

Forcing consumers to flatten + rename creates two sources of truth for the same images and ~2x disk for the test split.

## Detailed instructions for agents

**Files:** `src/raitap/data/data.py` (around lines 405-505).

**Behaviour to implement:**

1. Add an `id_strategy` config knob to the labels source config: one of `auto` (default), `relative_path`, `stem`.
2. In `auto` mode: if any value in the `image` column contains `/` or `\`, treat the entire column as relative paths; otherwise fall back to current stem behaviour (full backward-compat with existing flat-dir usage).
3. When in `relative_path` mode (or auto-detected as such): resolve each value as `(data.source / value).resolve()` and require the resulting file to exist. Strip extension only for the lookup key, not for the path resolution.
4. Symmetrical with `data.source` resolution rules — must work for both local dirs and URL-cached downloads.
5. Update `_normalise_sample_id` to be path-aware when strategy != stem.

**Tests to add:**

- Fixture with two-class nested layout, colliding stems, labels.csv using \`NORMAL/IM-0001.jpeg,0\` style. Assert all rows survive and align to correct images.
- Existing flat-dir tests must still pass under default \`auto\`.
- Explicit \`id_strategy: stem\` with nested layout → expect old behaviour (or a clear error).

**Config schema update:**

\`\`\`python
@dataclass
class LabelsConfig:
    source: str
    encoding: Literal[\"index\", \"one_hot\", \"argmax\"] = \"argmax\"
    id_strategy: Literal[\"auto\", \"relative_path\", \"stem\"] = \"auto\"
\`\`\`

## Consumer example (MLOps pipeline)

We want one canonical test directory serving both \`torchvision.datasets.ImageFolder\` (for training-time eval) and RAITAP:

\`\`\`
data/processed/clean/test/
├── NORMAL/IM-0001.jpeg
├── NORMAL/IM-0002.jpeg
├── PNEUMONIA/person1_bacteria_1.jpeg
└── PNEUMONIA/person1_bacteria_2.jpeg
\`\`\`

\`\`\`csv
image,label
NORMAL/IM-0001.jpeg,0
NORMAL/IM-0002.jpeg,0
PNEUMONIA/person1_bacteria_1.jpeg,1
PNEUMONIA/person1_bacteria_2.jpeg,1
\`\`\`

\`\`\`yaml
data:
  source: ./data/processed/clean/test
labels:
  source: ./data/processed/clean/labels.csv
  encoding: index
  # id_strategy: auto  ← detects \"/\" → uses relative paths
\`\`\`

Today this fails silently when stems collide. With this change, no flat-dir glue code is needed on the consumer side.
